src/main/java/net/mcreator/ui/laf/FileIcons.java

```diff
@@ -26,10 +26,10 @@
 public class FileIcons {
 
 	public static ImageIcon getIconForFile(File file) {
-		return getIconForFile(file.getName());
+		return getIconForFile(file.getName(), !file.isDirectory());
 	}
 
-	public static ImageIcon getIconForFile(String file) {
+	public static ImageIcon getIconForFile(String file, boolean leaf) {

```

Comment:
This will now still display a folder called "folder.java" with class icon and not with directory icon. Reorder conditions